x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialRelatesFunction.java

```diff
@@ -188,4 +194,94 @@ protected boolean pointRelatesGeometry(Point point, Component2D component2D) {
             }
         }
     }
+
+    /**
+     * When dealing with ST_CONTAINS and ST_WITHIN we need to pre-combine the field geometries for multi-values in order
+     * to perform the relationship check. This means instead of relying on the generated evaluators to iterate over all
```

Comment:
Why don't we use this approach for Intersects and Contains? I would rather prefer to use the same approacch in all functions and it will remove lots of complexity too.

Reply:
All functions now use the Block.Builder approach.